crates/status/src/status_manager.rs

```diff
@@ -16,93 +20,131 @@ pub enum StatusError {
         }
     }
 
+    /// Sends the updated `ClientState` to the client state receiver. Logs a warning if the receiver
+    /// is dropped.
     pub fn update_client_state(&self, post_state: &ClientState) {
         if self.cl.send(post_state.clone()).is_err() {
             warn!("client state receiver dropped");
         }
     }
 
+    /// Sends the updated `L1Status` to the L1 status receiver. Logs a warning if the receiver is
+    /// dropped.
     pub fn update_l1_status(&self, post_state: &L1Status) {
         if self.l1.send(post_state.clone()).is_err() {
             warn!("l1 status receiver dropped");
         }
     }
 
-    pub fn update_csm_status(&self, post_state: &CsmStatus) {
-        if self.csm.send(post_state.clone()).is_err() {
-            warn!("csm status receiver dropped");
-        }
+    /// Subscribes to updates for `ClientState` and returns a receiver.
+    pub fn subscribe_cl(&self) -> watch::Receiver<ClientState> {
+        self.cl.subscribe()
     }
 }
 
-/// Bundle wrapper for Status sender
-pub struct StatusTx {
-    pub csm: watch::Sender<CsmStatus>,
-    pub cl: watch::Sender<ClientState>,
-    pub l1: watch::Sender<L1Status>,
-    pub chs: watch::Sender<Option<Chainstate>>,
+impl StatusReceiver {
+    /// Gets the latest [`LocalL1State`].
+    pub fn l1_view(&self) -> LocalL1State {
+        self.cl.borrow().l1_view().clone()
+    }
+
+    /// Gets the latest [`SyncState`].
+    pub fn sync_state(&self) -> Option<SyncState> {
+        self.cl.borrow().sync().cloned()
+    }
+
+    /// Gets the latest operator table.
+    pub fn operator_table(&self) -> Option<OperatorTable> {
+        self.chs
+            .borrow()
+            .clone()
+            .map(|chs| chs.operator_table().clone())
+    }
+
+    /// Gets the latest deposits table.
+    pub fn deposits_table(&self) -> Option<DepositsTable> {
+        self.chs
+            .borrow()
+            .clone()
+            .map(|chs| chs.deposits_table().clone())
+    }
+
+    /// Gets the latest [`L1Status`].
+    pub fn l1_status(&self) -> L1Status {
+        self.l1.borrow().clone()
+    }
+
+    /// Waits until there's a new client state and returns the client state.
+    pub async fn wait_for_client_change(&self) -> Result<ClientState, RecvError> {
+        let mut s = self.cl.clone();
+        s.changed().await?;
+        let state = s.borrow().clone();
+        Ok(state)
+    }
+
+    /// Waits until genesis and returns the client state.
+    pub async fn wait_until_genesis(&self) -> Result<ClientState, RecvError> {
+        // Check if genesis has already occurred
+        let mut rx = self.cl.clone();
+        if rx.borrow().has_genesis_occurred() {
+            return Ok(self.cl.borrow().clone());
+        }
+
+        // If not wait for it
+        loop {
+            rx.changed().await?;
+            let state = rx.borrow();
+            if state.has_genesis_occurred() {
+                return Ok(state.clone());
+            }
```

Comment:
See if `.wait_for` does what we need here.

Reply:
Actually it doesn't because there's one more check besides changed() we need to do which is check if genesis happened. But this method can be simplified like this:
```rust
pub async fn wait_until_genesis(&self) -> Result<ClientState, RecvError> {
        let mut rx = self.cl.clone();
        loop {
            if rx.borrow().has_genesis_occurred() {
                return Ok(rx.borrow().clone());
            }
            rx.changed().await?;
        }
    }
```